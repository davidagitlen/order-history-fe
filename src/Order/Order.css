.order-card {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  min-height: 250px;
  width: 80%;
  margin: 10px 0px 10px 0px;
  background-color: rgba(248, 248, 255, .8);
}

.card-center {
  text-align: left;
  width: 55%;
}

.card-right p {
  font-size: 25px;
  font-weight: bold;
  margin: 0px 0px 25px 0px;
}

.order-card button {
  font-size: 22px;
  height: 100px;
  width: 175px;
  background-color: red;
  color: white;
}

.order-card button:hover {
  font-style: italic;
}

h2 {
  margin: 0px 0px 25px 0px;
}

img {
  height: 200px;
  width: 225px;
}


