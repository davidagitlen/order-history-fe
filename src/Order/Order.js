import React from 'react';
import './Order.css';

const Order = ({id, image, name, description, price, removePurchase}) => {
  
  return (
    <div className='order-card'>
      <div className='card-left'>
        <img src={image} alt='' />
      </div>
      <div className='card-center'>
        <h2>{name}</h2>
        <p>{description}</p>
      </div>
      <div className='card-right'>
        <p>${price}</p>
        <button onClick={() => removePurchase(id)}>
          Remove From History
          </button>
      </div>
    </div>
  )
}

export default Order;