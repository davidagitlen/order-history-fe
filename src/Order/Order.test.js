import React from 'react';
import { shallow } from 'enzyme';
import Order from './Order';

describe('Order', () => {
  let wrapper, removePurchaseMock;

  beforeEach(() => {
    removePurchaseMock = jest.fn()
    wrapper = shallow(<Order id={1} image='broken-link.png' name='order' description='my order' price={100} removePurchase={removePurchaseMock}/>);
  });

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call removePurchase prop when button is pushed with cards id', () => {
    wrapper.find('button').simulate('click');
    expect(removePurchaseMock).toHaveBeenCalledWith(1);
  })
})