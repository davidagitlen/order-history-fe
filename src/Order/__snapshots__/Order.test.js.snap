// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order should match the snapshot with all data passed in correctly 1`] = `
<div
  className="order-card"
>
  <div
    className="card-left"
  >
    <img
      alt=""
      src="broken-link.png"
    />
  </div>
  <div
    className="card-center"
  >
    <h2>
      order
    </h2>
    <p>
      my order
    </p>
  </div>
  <div
    className="card-right"
  >
    <p>
      $
      100
    </p>
    <button
      onClick={[Function]}
    >
      Remove From History
    </button>
  </div>
</div>
`;
